src/cmd/split.h

```diff
@@ -91,69 +92,72 @@ char* strdup_with_new(const char* the_string) {
 // See //util/csv/parser.h for more complete documentation.
 //
 // ----------------------------------------------------------------------
-void SplitCSVLineWithDelimiter(char* line, char delimiter, std::vector<char*>* cols) {
+void SplitLineWithDelimiter(char* line, const char* delimiter, std::vector<char*>* cols, const char enclosed) {
     char* end_of_line = line + strlen(line);
     char* end;
     char* start;
+    int delimiter_len = strlen(delimiter);
 
-    for (; line < end_of_line; line++) {
-        // Skip leading whitespace, unless said whitespace is the delimiter.
-        while (absl::ascii_isspace(*line) && *line != delimiter) ++line;
+    for (; line < end_of_line; line += delimiter_len) {
+        // Skip leading whitespace, unless said whitespace is the part of delimiter.
+        while (absl::ascii_isspace(*line) && *line != delimiter[0]) ++line;
 
-        if (*line == '"' && delimiter == ',') {  // Quoted value...
+        if (*line == enclosed) {  // Quoted value...
             start = ++line;
             end = start;
             for (; *line; line++) {
-                if (*line == '"') {
+                if (*line == enclosed) {
                     line++;
-                    if (*line != '"') {  // [""] is an escaped ["]
-                        break;           // but just ["] is end of value
+                    if (*line != enclosed) {  // [""] is an escaped ["]
+                        break;                // but just ["] is end of value
                     }
                 }
                 *end++ = *line;
             }
             // All characters after the closing quote and before the comma
             // are ignored.
-            line = strchr(line, delimiter);
+            line = strstr(line, delimiter);
             if (!line) line = end_of_line;
         } else {
             start = line;
-            line = strchr(line, delimiter);
+            line = strstr(line, delimiter);
             if (!line) line = end_of_line;
-            // Skip all trailing whitespace, unless said whitespace is the delimiter.
+            // Skip all trailing whitespace
             for (end = line; end > start; --end) {
-                if (!absl::ascii_isspace(end[-1]) || end[-1] == delimiter) break;
+                if (!absl::ascii_isspace(end[-1])) break;
             }
         }
-        const bool need_another_column = (*line == delimiter) && (line == end_of_line - 1);
         *end = '\0';
         cols->push_back(start);
         // If line was something like [paul,] (comma is the last character
         // and is not proceeded by whitespace or quote) then we are about
         // to eliminate the last column (which is empty). This would be
         // incorrect.
+        const bool need_another_column = (line + delimiter_len == end_of_line);
```

Comment:
We should check if `*line` is the first char of the delimiter to handle the case of the comment above.